sdk/core/Azure.Core/src/Serialization/ModelSerializerFormat.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// A format used to specify how a model should be serialized and deserialized.
+    /// </summary>
+    public readonly partial struct ModelSerializerFormat : IEquatable<ModelSerializerFormat>
+    {
+        internal const string JsonValue = "J";
+        internal const string WireValue = "W";
+
+        /// <summary>
+        /// Default format which will serialize all properties including read-only and additional properties.
+        /// The format will always be JSON.
+        /// </summary>
+        public static readonly ModelSerializerFormat Json = new ModelSerializerFormat(JsonValue);
+
+        /// <summary>
+        /// Format used to communicate with Azure services.  This format will not serialize read-only properties or additional properties.
+        /// The format will vary between XML and JSON depdending on the service.
+        /// </summary>
+        public static readonly ModelSerializerFormat Wire = new ModelSerializerFormat(WireValue);
```

Comment:
What is the difference between JSON and Wire?  It might be helpful to add a discussion of formats to the samples.

Reply:
Since you mention it, do we want to use "wire" anyway? I don't think that will be well-understood to all developers given background, cultures, etc. We're in a wireless age, so I can anticipate confusion. Maybe something like "SourceFormat" or "ServiceFormat" or something more specific and obvious?

Reply:
Does the comment here not describe what the difference is?  What do you think is missing from these comments?

@tg-msft as he had some concerns around this documentation as well?

+1 to the samples

Reply:
> What do you think is missing from these comments?

As a user of these APIs, it's not clear to me whether I should pass `Json` or `Wire` - when would I chose one or the other and why?

Reply:
Yeah, I pitched a fit for a good long while about this during the Arch Board, but I was the only one sufficiently bothered by it.  :smile:  The final consensus was we'd make `Wire` look necessarily complicated/worse than all the other values like `Json`, (maybe) `Xml`, `JsonPatch`, `JsonMergePatch`, etc.

I think we should make the docs aggressively clear that it's THE input/request format and less likely to be useful for general purpose serialization.  Maybe something like:
```C#
/// Format used to serialize this model when sending as a request to an Azure service.
/// It will not serialize read-only properties or additional properties.
/// The content-type will vary between JSON, XML, etc., depending on the service.
/// 
/// Most use cases should prefer a more complete format like `Json` that includes
/// read-only and additional properties.
```

Reply:
+@KrzysztofCwalina as well in case he has thoughts about how we should document this.

Reply:
If it is intended primarily for our own use when serializing models to send to services as input parameters, would it make sense to name it something a bit longer to scare people off?  For example `ServiceInput`?  Wire could mean something transmitted in any direction.

Reply:
going to stick with Wire and Json and add the xml docs that Ted suggested above.